packages/adminrouter/extra/src/lib/service.lua

```diff
@@ -312,6 +312,7 @@ local function recursive_resolve(auth, path, marathon_cache, mesos_cache)
     local do_request_buffering = true
     local err_code = nil
     local err_text = nil
+    local service_name = ""
```

Comment:
Next time it would be nice to add a code comment here stating the importance of this declaration.

(the non-obvious part here is that things "seem to work" even without this declaration)

Reply:
@jgehrcke There are backport foreseen for 1.11 and 1.10: https://jira.mesosphere.com/browse/DCOS_OSS-4999
Does it make sense to add it to 1.10's CHANGES.md?

Reply:
I asked @ahairape to remove the comment that he initially added. Declaring variables as local should be the default expectation. All these local variables could cause problems if they are not declared local. We should add an explicit comment when something is _not_ local. (or Lua should default to local unless a variable is global - viva Python!)